City-of-Helsinki/drupal-helfi-etusivu
# UHF-8395 local task block translation

# [UHF-8395 ](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8395 )
"Local tasks" is the block show for editors and admins above content containing view, edit, translate, revision (etc.) links.
Local tasks block in content page used to follow content language. This caused links such as edit / translate / view to be translated in different language and in case of RTL language, switch the order as well.

[Related Drupal Core issue](https://www.drupal.org/project/drupal/issues/3054641)

## What was done
* Created new block based on core's LocalTasksBlock.
* Override the link titles based on user preference (preferred admin language)
* Added user as cache context

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-8395`
  * `make fresh`
* Run `make drush-cr`

## How to test
- Log in as admin
- Add a translation to existing page on RTL language, for example arabia
- Select 2 users (with permission to edit ther own account settings), check out their user id's
- Login as first one (drush uli --uid={user_id_goes_here})
- Edit user's preferred admin language as (anything else than fi, en, sv)
- Go to the page you just translated. You should see the local tasks menu links in language set as your preferred language
- Log out
- Log in as the second user, and change his preferred language to something else than the last user had
- Go to the translated page, same page as last time: You should see the local tasks menu links in language set as your preferred language. Language other than the first user saw. This means caching works as it should
